src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -479,14 +479,14 @@ public virtual string Href([NotNull] string contentPath)
             return _urlHelper.Content(contentPath);
         }
 
-        private void WritePositionTaggedLiteral(TextWriter writer, string value, int position)
+        private void WritePositionTaggedLiteral([NotNull] TextWriter writer, [NotNull] string value, int position)
```

Comment:
Overkill for the `TextWriter` here, method is `private` and call sites all have `[NotNull]` for all `TextWriter` bits. Don't feel strongly though, up to you if you want to keep it.